### ISSUE TITLE
Customizing glossary terms appearence

### ISSUE THREAD

**Issue by elenaferr0** [swaits/typst-collection#2]
Hi, I'm looking for a way to customize the look of glossary terms. Specifically, I'd like to show a superscript or subscript on words which are in the glossarium, for example:
```typst
// In the typst file
Lorem @ipsum ...
```
and the output would be
$Lorem  ipsum^{g}$ so that the reader knows that when a word has the superscript `[g]` it can be found in the glossary.
I'm aware that customizing the appearance of terms is possible by changing the show rule for links like follows:
```typst
#show link: set text(fill: blue.darken(60%))
```
but I don't want to change it for every link (it would not really make sense if a link to a web page had `[g]`).

Do you know by chance if there is any way of doing this? I tried to search for workarounds too but I did not happen to find anything. I'm not an expert about typst so that might also be the cause. Thank you in advance

**Comment by swaits** [comment]
Interesting request.

One way to add a feature for this in `glossy` might be to add a parameter to this:

```typst
// Main wrapper (usually used in a `#show: init-glossary`) which loads the
// entries passed in into our state. Furthermore, hooks into references so that
// we can intercept term usage in a doc and label them appropriately.
#let init-glossary(entries, body) = context {
  for entry in __normalize_entries(entries) {
    __add_entry(entry)
  }

  // convert refs we recognize into links with labels
  show ref: r => {
    let (key, ..modifiers) = str(r.target).split(":")
    if __has_entry(key) {
      __gls(key, modifiers: modifiers)
    } else {
      r
    }
  }

  body
}
```

So it'd be something like `init-glossary(entries, show-term: none, body)`. This would let the user do:

```typst
#let my-show-term(body) = [ #body#super[g] ]

#show: init-glossary.with(my-glossary, show-term: my-show-term)

// rest of doc follows
```

I'm open to other suggestions for the interface. Pending such suggestions, if you like this let me know and I can add it in the next version of `glossy`.

**Comment by swaits** [comment]
I went ahead and added this to 0.1.2. The PR to add it to the main repo is https://github.com/typst/packages/pull/1209

Once that gets merged, you should be all set. The README explains how to use it. But it's exactly as I described above.

**Comment by elenaferr0** [comment]
Oh super, thank you very much!

**Comment by swaits** [comment]
You're very welcome. Looks like it got merged so 0.1.2 should be live now.
